_rules/links-with-identical-names-and-context-serve-equivalent-purpose-fd3a94.md

```diff
@@ -229,80 +243,124 @@ These two HTML `a` elements have the same [accessible name][] and [context][prog
+	</p>
+</html>
+```
+
 #### Failed Example 2
 
-These two HTML `a` elements have the same [accessible name][] and [context][programmatically determined link context]. They link to web pages that are similar, but have different information in their content.
+These two HTML `a` elements have the same [accessible name][] and [context][programmatically determined link context], but go to different resources. Their purpose is disambiguated for sighted users by the alignment of the links with the images above.
 
 ```html
 <html lang="en">
 	<div>
-		Learn more (<a href="/test-assets/links-with-identical-names-serve-equivalent-purpose-b20e66/about/contact.html"
-			>Contact us</a
-		>) and get in touch (
-		<a href="/test-assets/links-with-identical-names-serve-equivalent-purpose-b20e66/admissions/contact.html"
-			>Contact us</a
-		>)
+		<span style="text-align:center;">Contact us</span>
+		<span style="display:flex; justify-content:space-around;">
+			<img src="/test-assets/shared/chat.png" alt="Chat" style="width:50%;">
+			<img src="/test-assets/shared/phone.png" alt="Phone" style="width:50%;">
+		</span>
+		<span style="display:flex; justify-content:space-around;">
+			<a href="/test-assets/links-with-identical-names-serve-equivalent-purpose-b20e66/contact-us.html?page=60e950cff70bf1ec60a702086748ab4dec361514">Contact Us</a>
+			<a href="/test-assets/links-with-identical-names-serve-equivalent-purpose-b20e66/contact-us.html?page=c1d4e0f067462f4b28716f028d9213a25eb82f28">Contact Us</a>
+		</span>
 	</div>
 </html>
 ```
 
 #### Failed Example 3
 
-These two HTML `span` elements have an [explicit role][] of link, same [accessible name][] and [context][programmatically determined link context], but link to resources that offer different content.
+These two HTML `a` elements have the same [accessible name][] and [context][programmatically determined link context]. They link to web pages that are similar, but have different information in their content. Their purpose is disambiguated for sighted users by the alignment of the links with the images above.
 
 ```html
 <html lang="en">
-	<p>
-		Learn more (<span
+	<div>
+		<span style="text-align:center;">Contact us</span>
+		<span style="display:flex; justify-content:space-around;">
+			<img src="/test-assets/shared/chat.png" alt="Chat" style="width:50%;">
+			<img src="/test-assets/shared/phone.png" alt="Phone" style="width:50%;">
+		</span>
+		<span style="display:flex; justify-content:space-around;">
+			<a href="/test-assets/links-with-identical-names-serve-equivalent-purpose-b20e66/contact-us.html?page=3">Contact Us</a>
+			<a href="/test-assets/links-with-identical-names-serve-equivalent-purpose-b20e66/contact-us.html?page=4">Contact Us</a>
+		</span>
+	</div>
+</html>
+```
```

Comment:
I'm not sure I remember why we kept both example 2 and 3 😓 
They seem to be the same, except for the query parameters in the URLs 🤔

Reply:
- example 2 refers to completely different resources (?page=1 and ?page=2);
- example 3 refers to similar resources with the same page structure but different values for specific fields (page=3 and ?page=4 e.g. phone number and opening hours).

This distinction was already present at the time of the edits, so I kept it also for the new adjustments. If it's not necessary we can remove it.

Reply:
Ah, yes, thanks for the refresh.
Then, I guess I'm challenging the complex query parameter which seems unnecessarily and totally took me off when looking at them.
I do remember we did discuss these as an example of "the URL is not always enough to disambiguate". But in the end this is not something that is present in the rule (mentioned in the assumption, but not really discussed in the rule). I guess example 2 would work as well with `page1` and `page2` as query parameters 🤔 (it would also simplify the code by removing the test). But OTOH, we may want to keep a "realistic" hash query parameter.

Reply:
@Jym77, 
I agree with above.
Maybe a more clear understanding (page1 and page2 as query params) is better than covering a "realistic" case but both the solutions work for me.

Reply:
Let's see if others have opinions about it 😃